Tools/Scripts/hooks/prepare-commit-msg

```diff
@@ -290,6 +288,34 @@ def cherry_pick(content):
             result.append(line)
     return '\n'.join(result)
 
+def revert(source=None, sha=None):
+    commit_message = []
+    command = [{{ perl }}, os.path.join(SCRIPTS, 'prepare-ChangeLog'), '--no-write', '--only-files', '--delimiters', '--git-index']
+
+    if sha:
+        commit_message.append('Amend changes:')
+        parseChanges(command, commit_message)
+        commit_message.append('')
+
+        commit_message.append('Combined changes:')
+        command.append('--git-commit')
+        command.append('HEAD')
+
+    parseChanges(command, commit_message)
+
+    bugs_string = get_bugs_string()
+
+    return '''{title}
+{revert_msg}
+{bugs}
+
+{content}
+'''.format(
+        title=os.environ.get('COMMIT_MESSAGE_TITLE', '') or 'Need a short description (OOPS!).',
```

Comment:
You should probably have a default here that basically amounts to "Revert <commit ref>"